File: solutions/3075. Maximize Happiness of Selected Children/3075.py
# Time complexity: O(nlogn)
# Space complexity: O(1)
class Solution:
  def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
    happiness.sort(reverse=True)
    count = 0
    hSum = 0

    for i in range(k):
      if happiness[i] - count < 0:
        return hSum

      hSum += happiness[i] - count
      count += 1

    return hSum

File: solutions/3075. Maximize Happiness of Selected Children/3075.js
// Time complexity: O(nlogn)
// Space complexity: O(1)
var maximumHappinessSum = function(happiness, k) {
  happiness.sort((a, b) => b - a);
  let count = 0;
  let sum = 0;

  for (let i = 0; i < k; ++i) {
    if (happiness[i] - count < 0)
      return sum;

    sum += happiness[i] - count;
    ++count;    
  }

  return sum;
};